app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,8 @@
+class Admin::UsersController < Admin::ApplicationController
+  def switch_user
+    user = User.find params[:user_id]
+    sign_in(:user, user)
+    flash[:notice] = "You are now signed in as #{user.principal.full_name}"
```

Comment:
Actually, the admin stuff isn't internationalized at all is it? In which case leave it I guess!